examples/user_guide/Linking_Plots.ipynb

```diff
@@ -0,0 +1,220 @@
+   "source": [
+    "renderer = hv.renderer('bokeh')\n",
+    "\n",
+    "plot = renderer.get_plot(hv.Scatter([]))\n",
+    "\n",
+    "plot.handles.keys()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this case we are interested in the 'source' handle, but we still have to tell it which events should trigger the callback. Bokeh callbacks can be grouped into two types model property changes and and events. For more detail on these two types of callbacks see the [Bokeh user guide](https://bokeh.pydata.org/en/latest/docs/user_guide/interaction/callbacks.html#userguide-interaction-jscallbacks).\n",
+    "\n",
+    "In this case we want to respond to changes to the ``ColumnDataSource.selected`` property. We can declare this in the ``on_source_changes`` class atttribute on our callback. So now that we have declared which model we want to listen to events on and which events we want to listen to we have to declare the model on the target we want to change in response. \n",
+    "\n",
+    "We can once again look at the handles on the plot corresponding to the ``HLine`` element:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot = renderer.get_plot(hv.HLine(0))\n",
+    "plot.handles.keys()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now want to change the ``glyph``, which defines the position of the ``HLine``, so we declare the ``target_model`` as ``'glyph'``. Having defined both the source and target model and the events we can finally start writing the JS callback that should be triggered. To declare it we simply define the ``source_code`` class attribute. To understand how to write this code we need to understand how the source and target models we have declared can be referenced from within the callback.\n",
+    "\n",
+    "The ``source_model`` will be made available by prefixing it with ``source_``, while the target model is made available with the prefix ``target_``. This means that the ``ColumnDataSource`` on the ``source`` can be referenced as ``source_source``, while the glyph on the target can be referenced as ``target_glyph``.\n",
+    "\n",
+    "Once we know how to reference the models we can access their properties to compute the mean y-value of the current selection on the source ``ColumnDataSource`` and set the ``target_glyph.position`` to that value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from holoviews.plotting.bokeh.callbacks import LinkCallback\n",
+    "\n",
+    "class MeanHLineCallback(LinkCallback):\n",
+    "\n",
+    "    source_model = 'source'\n",
+    "    on_source_changes = ['selected']\n",
+    "    \n",
+    "    target_model = 'glyph'\n",
+    "    \n",
+    "    source_code = \"\"\"\n",
+    "        var inds = source_source.selected.indices;\n",
```

Comment:
Ok, so ``source_source`` is a real thing. I *think* I'm getting the idea but I am finding this rather confusing at the moment.

Reply:
It's very similar to the way streams work, but I'll have another go at clarifying this.